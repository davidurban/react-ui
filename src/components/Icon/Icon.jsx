import PropTypes from 'prop-types';
import React from 'react';
import styles from './Icon.scss';
import loadMaterialDesignIcons from './load-material-design-icons';

loadMaterialDesignIcons();

const Icon = (props) => {
  let iconClass = styles.root;

  if (props.size === 'large') {
    iconClass = styles.isRootLarge;
  } else if (props.size === 'larger') {
    iconClass = styles.isRootLarger;
  }

  return (
    <svg
      style={{ fill: 'currentColor' }}
      className={iconClass}
    >
      <use xlinkHref={`#ic_${props.icon}_48px`} />
    </svg>
  );
};

Icon.defaultProps = {
  size: 'default',
};

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  size: PropTypes.oneOf(['default', 'large', 'larger']),
};

export default Icon;
