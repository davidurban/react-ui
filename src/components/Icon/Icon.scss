.root {
  display: inline-block;
  width: 1rem;
  height: 1rem;
}

.isRootLarge {
  composes: root;
  width: 1.5rem;
  height: 1.5rem;
}

.isRootLarger {
  composes: root;
  width: 2rem;
  height: 2rem;
}
