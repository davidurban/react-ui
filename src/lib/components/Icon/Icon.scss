.root {
  display: inline-block;
  width: 1rem;
  height: 1rem;
}

.isRootSmall {
  composes: root;
  width: 0.75rem;
  height: 0.75rem;
}

.isRootLarge {
  composes: root;
  width: 1.5rem;
  height: 1.5rem;
}

.isRootLarger {
  composes: root;
  width: 2rem;
  height: 2rem;
}
