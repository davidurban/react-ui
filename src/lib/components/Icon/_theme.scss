$icon-size-default: var(--rui-icon-size-default);
$icon-size-small: var(--rui-icon-size-small);
$icon-size-large: var(--rui-icon-size-large);
$icon-size-larger: var(--rui-icon-size-larger);
