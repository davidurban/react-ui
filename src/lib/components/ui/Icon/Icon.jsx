import PropTypes from 'prop-types';
import React from 'react';
import styles from './Icon.scss';
import loadMaterialDesignIcons from './load-material-design-icons';

loadMaterialDesignIcons();

const Icon = (props) => {
  let iconClass = styles.root;

  if (props.size === 'small') {
    iconClass = styles.rootSizeSmall;
  } else if (props.size === 'large') {
    iconClass = styles.rootSizeLarge;
  } else if (props.size === 'larger') {
    iconClass = styles.rootSizeLarger;
  }

  return (
    <svg
      className={iconClass}
      id={props.id}
      style={{ fill: 'currentColor' }}
    >
      <use xlinkHref={`#ic_${props.icon}_48px`} />
    </svg>
  );
};

Icon.defaultProps = {
  id: undefined,
  size: 'medium',
};

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  id: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large', 'larger']),
};

export default Icon;
