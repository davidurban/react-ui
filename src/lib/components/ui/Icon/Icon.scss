@import 'theme';

.root {
  display: inline-block;
  width: $icon-size-default;
  height: $icon-size-default;
}

.rootSizeSmall {
  composes: root;
  width: $icon-size-small;
  height: $icon-size-small;
}

.rootSizeLarge {
  composes: root;
  width: $icon-size-large;
  height: $icon-size-large;
}

.rootSizeLarger {
  composes: root;
  width: $icon-size-larger;
  height: $icon-size-larger;
}
