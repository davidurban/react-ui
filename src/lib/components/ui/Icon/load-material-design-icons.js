export default () => {
  const importAll = (r) => {
    r.keys().forEach((key) => { r(key); });
  };

  importAll(require.context('!svg-sprite-loader!material-design-icons/', true, /[\\/]svg[\\/]production[\\/]ic_.*_48px\.svg$/));
};
