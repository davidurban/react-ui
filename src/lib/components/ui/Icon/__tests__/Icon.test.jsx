import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Icon from '../';

jest.mock('../load-material-design-icons');

describe('rendering', () => {
  it('renders correctly mandatory props only', () => {
    const tree = shallow(<Icon
      icon="book"
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with size', () => {
    const tree = shallow(<Icon
      icon="book"
      size="larger"
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
