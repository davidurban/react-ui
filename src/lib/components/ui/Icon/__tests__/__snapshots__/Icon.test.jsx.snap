// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly mandatory props only 1`] = `
<svg
  className="root"
  style={
    Object {
      "fill": "currentColor",
    }
  }
>
  <use
    xlinkHref="#ic_book_48px"
  />
</svg>
`;

exports[`rendering renders correctly with all props 1`] = `
<svg
  className="rootSizeLarger"
  id="custom-id"
  style={
    Object {
      "fill": "currentColor",
    }
  }
>
  <use
    xlinkHref="#ic_book_48px"
  />
</svg>
`;
